src/som/interpreter/actors/EventualMessage.java

```diff
@@ -233,13 +245,15 @@ public SPromise getPromise() {
   /** The callback message to be send after a promise is resolved. */
   public static final class PromiseCallbackMessage extends PromiseMessage {
     protected final SPromise promise;
+    protected final SPromise originalPromise;
```

Comment:
I put the clarification as a comment in getOriginalPromise() method: The promise on which this callback is registered on. I need it to check if a stepReturn was trigger for that promise.